Assets/SEE/VCS/GitVersionControl.cs

```diff
@@ -0,0 +1,187 @@
+        /// The repository.
+        /// </summary>
+        private readonly Repository repo;
+
+        /// <summary>
+        /// The path to the root of the repository.
+        /// </summary>
+        private readonly string repositoryPath;
+
+        /// <summary>
+        /// See <see cref="IVersionControl.Show(string, string)"/>.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">throw in case <paramref name="fileName"/>
+        /// or <paramref name="commitID"/> are null or just whitespace</exception>
+        public string Show(string fileName, string commitID)
+        {
+            if (string.IsNullOrWhiteSpace(fileName))
+            {
+                throw new ArgumentNullException("Path must not be null or empty.");
+            }
+            if (string.IsNullOrWhiteSpace(commitID))
+            {
+                throw new ArgumentNullException("Commit ID must not be null or empty.");
+            }
+            Blob blob = repo.Lookup<Blob>($"{commitID}:{Path.GetRelativePath(repositoryPath, fileName).Replace("\\", "/")}");
+
+            if (blob != null)
+            {
+                return blob.GetContentText();
+            }
+            else
+            {
+                // File does not exist.
+                return "";
+            }
+        }
+
+        /// <summary>
+        /// See <see cref="IVersionControl.GetFileChange(string, string, string, out string)"/>.
+        /// </summary>
+        public Change GetFileChange(string fileName, string oldCommitID, string newCommitID, out string oldFilename)
+        {
+            Commit newCommit = repo.Lookup<Commit>(newCommitID);
+            if (newCommit == null)
+            {
+                throw new UnknownCommitID($"Unknown commit id {newCommit} for git repository at {repositoryPath}");
+            }
+
+            Commit oldCommit = repo.Lookup<Commit>(oldCommitID);
+            if (oldCommit == null)
+            {
+                throw new UnknownCommitID($"Unknown commit id {oldCommit} for git repository at {repositoryPath}");
+            }
+
+            CompareOptions compareOptions = new()
+            {
+                Algorithm = DiffAlgorithm.Myers,
+                Similarity = new SimilarityOptions
+                {
+                    RenameDetectionMode = RenameDetectionMode.Default,
+                }
+            };
+
+            // Compare the commits
+            IEnumerable<TreeEntryChanges> changes = repo.Diff.Compare<TreeChanges>(oldCommit.Tree, newCommit.Tree, compareOptions)
+                .Where(change => change.Path == fileName || change.OldPath == fileName);
+
+            Change result = Change.Unknown;
+            oldFilename = null;
+
+            int numberOfIterations = 0;
+            // Note: changes.Count can only be 0 (no change what so ever) or 1.
+            // We are still using a loop to obtain the change even though the loop
+            // has only one iteration at most.
```

Comment:
Why is a for loop used, though? To assert that only 1 or 0 elements is present in an enumerable, you can use `Change change = changes.SingleOrDefault()`, which returns null for 0 elements, the element itself for 1 element, and throws an exception in all other cases.

Reply:
@lilSander was the original author of this loop. I am actually not sure yet, whether there could be more than one iteration. That is why I added a check (assertion) after the loop. But we can definitely have 0 iterations.

I created and linked an issue #703.